app/src/main/java/com/pose/move/feature/profile/menu/MenuViewModel.kt

```diff
@@ -21,7 +21,7 @@ class MenuViewModel @Inject constructor(
             if (clearPhoto) {
                 internalStorageManager.setHasUserAddedLicensePhoto(false)
             }
-            internalStorageManager.setIsUserLoggedIn(false)
+            internalStorageManager.setAuthToken("")
```

Comment:
I'd suggest moving this into an usecase as well and not scoping it to the viewModel.
You should use a app-wide scope for this (like GlobalScope.launch, but i'd suggest a custom one).
If you navigate away from this, it will get cancelled.